File: Beakjoon/문제풀이/일곱 난쟁이/jeoeo.py
heights = []
for _ in range(9):
    heights.append(int(input()))

total_sum = sum(heights)
false_sum = total_sum - 100

dwaf = False
for i in range(9):
    for j in range(i + 1, 9):
        if heights[i] + heights[j] == false_sum:
            # 가짜 난쟁이 제거
            height1, height2 = heights[i], heights[j]
            heights.remove(height1)
            heights.remove(height2)
            dwaf = True
            break
    if dwaf:
        break

heights.sort()
for height in heights:
    print(height)

File: Beakjoon/문제풀이/일곱 난쟁이/jeoeo.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.Scanner;

public class jeoeo {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        ArrayList<Integer> heights = new ArrayList<>();

        for (int i = 0; i < 9; i++) {
            heights.add(sc.nextInt());
        }

        int totalSum = 0;
        for (int height : heights) {
            totalSum += height;
        }

        int falseSum = totalSum - 100;

        boolean dwaf = false;
        // 두 난쟁이 찾기
        for (int i = 0; i < heights.size(); i++) {
            for (int j = i + 1; j < heights.size(); j++) {
                if (heights.get(i) + heights.get(j) == falseSum) {
                    heights.remove(j);
                    heights.remove(i);
                    dwaf = true;
                    break;
                }
            }
            if (dwaf) {
                break;
            }
        }

        Collections.sort(heights);

        for (int height : heights) {
            System.out.println(height);
        }

        sc.close();
    }
}
